library/math/dual.lisp

```diff
@@ -0,0 +1,135 @@
+;;;; dual.lisp
+;;;;
+;;;; dual numbers to enable automatic differentiation
+
+(coalton-library/utils:defstdlib-package #:coalton-library/math/dual
+    (:use
+     #:coalton
+     #:coalton-library/builtin
+     #:coalton-library/classes
+     #:coalton-library/functions
+     #:coalton-library/math/arith
+     #:coalton-library/math/elementary
+     #:coalton-library/math/integral)
+  (:export
+   #:Dual
+   #:primal-part
+   #:dual-part))
+
+(in-package #:coalton-library/math/dual)
+(named-readtables:in-readtable coalton:coalton)
+
+(coalton-toplevel
+
+  (define-type (Dual :t)
+    "Representation of a Dual number in the form (a + bE)."
+    (Dual :t :t))
+  
+  (declare primal-part (Dual :t -> :t))
+  (define (primal-part (Dual p _))
```

Comment:
I think your emacs indentation is misconfigured. Forms following a define should be indented two spaces not four. You might need [something like this](https://github.com/eliaslfox/dotfiles/blob/main/nixos/users/elf/files/init.el#L113-L114) to configure it.

Reply:
the link is broken.

Reply:
Not sure what's going on. I'll just copy and paste. My slime use-package looks like:

```
(use-package slime
    :ensure t
    :config
    (progn
        (load (expand-file-name "~/.local/share/quicklisp/slime-helper.el"))
        (setq lisp-indent-function 'common-lisp-indent-function)
        (push 'slime-indentation slime-contribs)
        (setq inferior-lisp-program "sbcl --noinform")))
```

My guess is that you need the lines mentioning indentation to have eMacs format lisp as Common Lisp.

Reply:
Ok - I have just pushed. I had to manually indent. I added your config to my `.emacs` but indentation was off a little still.